### PR TITLE
Fix validator and add notebooks and document for level-up validator

### PR DESCRIPTION
**Pull request by wonjuleee**
### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added the description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md).​
- [ ] I have updated the [documentation](https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly

### License

- [ ] I submit _my code changes_ under the same [MIT License](https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: docs/source/docs/level-up/intermediate_skills/08_data_validate.rst [hunk 1]
```diff
@@ -0,0 +1,70 @@
+=============
+Level 8: Dataset Validation
+=============
```
Comment (yunchu) [inline docs/source/docs/level-up/intermediate_skills/08_data_validate.rst]: different upper/lower lines with the title cause the build warning. it's better to have same length with title.

Comment (wonjuleee) [reply docs/source/docs/level-up/intermediate_skills/08_data_validate.rst]: Thank you for pointing out this. I updated.

File: datumaro/plugins/validators.py
Comment (vinnamkim) [inline datumaro/plugins/validators.py:1]: Would you tell us what you aimed to achieve by changing the code in this file? Just for refactoring?

File: tests/unit/test_validator.py [hunk 2]
```diff
@@ -405,10 +405,14 @@ def test_check_missing_attribute(self):
...
-        label_name = "unittest"
-        label_stats = {"items_with_undefined_label": [(1, "unittest")]}
+        label_name = "cat0"
+        item_id = 1
+        item_subset = "unittest"
+        label_stats = {label_name: {"items_with_undefined_label": [(item_id, item_subset)]}}
+        stats = {"label_distribution": {"undefined_labels": label_stats}}
 
-        actual_reports = self.validator._check_undefined_label(label_name, label_stats)
+        # actual_reports = self.validator._check_undefined_label(label_name, label_stats)
```
Comment (vinnamkim) [inline tests/unit/test_validator.py]: It seems to be removed.

File: datumaro/plugins/validators.py
Comment (wonjuleee) [reply datumaro/plugins/validators.py:1]: Yes I did, I tried to reuse some functions by refactoring.

File: tests/unit/test_validator.py (hunk 2 above)
Comment (wonjuleee) [reply tests/unit/test_validator.py]: I removed.

**Review by vinnamkim — APPROVED** [review]
LGTM.

**Review by sooahleex — APPROVED** [review]

**Pull request merged**